float64 deltaBi
float64 FrSpin